### DISCUSSION TITLE
Retrieval of just plastic section modulus in composite section

### DISCUSSION THREAD

**Discussion by mag06998** [robbievanleeuwen/section-properties#432] [Q&A] [answered]
I've been using this library to provide section properties for structural engineers at my company. Works great, but I've been asked to provide JUST the plastic section modulus rather than the plastic moment for whatever reason. This is easy enough to do when we have a shape with one material, but it's not immediately obvious to me how to get this number when we have composite sections with multiple materials, as the .get_mp() returns the plastic section modulus mulitplied by some yield strength. 

How is this yield strength calculated? My initial guess was that this yield strength is some effective number for the shape based on how much of each material is used and their respective yield strengths, but there's no obvious way to retrieve this number like there is with the effective young's modulus through .get_e_eff()... but maybe I'm missing something.

**Comment by robbievanleeuwen** [comment, marked as answer]
Hi @mag06998, if you are interested in some reading here are some relevant links in the documentation:

- [Geometric only analysis plastic theory](https://sectionproperties.readthedocs.io/en/stable/user_guide/theory.html#plastic-section-moduli)
- [Plastic analysis assumptions](https://sectionproperties.readthedocs.io/en/stable/user_guide/analysis.html#plastic-analysis)
- [How Material Properties Affect Results](https://sectionproperties.readthedocs.io/en/stable/user_guide/results.html#how-material-properties-affect-results)

I'll answer the second part of your question first - performing a plastic analysis of a composite section in `sectionproperties` assumes that all the material in the cross-section is experiencing a stress equal to its yield strength - this is consistent with the theory (note non-linear behaviour is not considered as per the documentation).  As such there is no effective yield strength, it is not calculated - it is supplied by you, the user, for each material you define. 

In simplistic terms, the calculation essentially involves multiplying the areas of each material by its yield strength to get a plastic force for each material. The centroid of these forces is then calculated (plastic centroid). Finally, the resultant moment these forces exert about the plastic centroid is determined, which is the plastic moment returned by `get_mp()`.

This leads to the first part of your question - there is *no definition* of a plastic section modulus for a composite section as this is purely a geometric property. Perhaps some kind of effective plastic section modulus could be contrived based on weighting the areas based on the yield strengths, but I would be interested in understanding what the application of this would be as I have never come across the need for this in my experience. It feels like the answer to this part of the question really relies on what the use of this value is for. If you are purely interested in the plastic section modulus of the shape/geometry, then I would suggest not supplying material properties at all (essentially analysing a homogenous section).

I hope the above is helpful, happy to discuss this further if you have more to add :)

**Reply by mag06998** [reply]
Thanks for the quick reply Robbie, makes sense! Appreciate the break down of how the plastic moment for composites is calculated, I was off on how I had been conceptualizing that.

No more to add right now, you've seemed to answer this well enough I can move forward. Since there is no definition for a plastic section modulus for a composite, I'll communicate that to them and just keep the plastic moment as the output when it's not a simple section. 

As for the use I'll have to ask. I think it really just comes down to the fact that the Mathcad sheets they have been using to do some of their calcs take it as an input. What information the plastic section provides that the plastic moment wouldn't suffice for I'm not sure - most of the time the extrusions are simple sections, which they would previously get the plastic section modulus for easily using shape builder software, then plug into the sheets. What it seems they would do in the past is make approximations for the parts of the composite sections that were different materials and plug them into the software as one material.. which certainly seems less accurate than what this library provides.

**Reply by EngDM** [reply]
@mag06998 

I am curious as to what they are using the plastic section modulus for, that the plastic moment would not work in place of with a slight adjustment to inputs. Are you able to elaborate what they use it for? I can't think of an application where plastic modulus gets used where it isn't immediately multiplied by yield strength and used as Mp.
